packages/react-dom/src/server/ReactPartialRendererHooks.js

```diff
@@ -495,15 +500,18 @@ function useTransition(
 }
 
 function useOpaqueIdentifier(): OpaqueIDType {
-  return makeServerId();
+  return (
+    (currentPartialRenderer.identifierPrefix || '') +
+    'R:' +
+    (currentPartialRenderer.uniqueID++).toString(36)
+  );
 }
 
 function noop(): void {}
 
-export let currentThreadID: ThreadID = 0;
-
-export function setCurrentThreadID(threadID: ThreadID) {
-  currentThreadID = threadID;
+export let currentPartialRenderer = {};
```

Comment:
This type isn't valid. I think it just accidental works since it's an "open type".

Set this to null so it fails if something goes wrong. Let's fool Flow though so you don't need null checks.

```
export let currentPartialRenderer: PartialRenderer = (null: any);
```